.github/workflows/ci.yaml

```diff
@@ -0,0 +1,86 @@
+name: CI
+on:
+  pull_request:
+  push:
+    branches: [master]
+  workflow_dispatch:
+    inputs:
+      debug_enabled:
+        description: "Run the build with tmate debugging enabled"
+        required: false
+jobs:
+  tox:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        tox-env: [lint, docs, check-packaging]
+    steps:
+      - name: Cancel previous workflows that are still running
+        uses: styfle/cancel-workflow-action@0.8.0
+        with:
+          access_token: ${{ github.token }}
+      - name: Checkout latest commit
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # fetch all history with version tags
```

Comment:
What is our use case for this new element?

Reply:
Yes, thanks from bringing this up.

The idea is that we use `setuptools-scm` to generate our version from git tags, but with the default shallow clone there will be not tags, and so the generated version will always be some default (`0.1` I believe).

I don't think this causes any problems for us, I saw this being a problem elsewhere, where the tests relied on the presence of the patch version, which is otherwise missing in the default version.

So all-in-all, for us this is not strictly necessary. On the other hand this step takes 2-3 seconds, so not a big deal either. What do you guys think?

Reply:
Fair enough. OK to have this new configuration. Resolving.